include/ttmlir/Dialect/TTIR/Analysis/TTIRAnalysis.h

```diff
@@ -41,7 +44,11 @@ template <class I, class R> class TTIRAnalysis {
   // Get the analysis result.
   //
   const R &getResult() {
-    runAnalysis();
+    auto overrideMap = tt::OverrideMap::getInstance();
+    if (overrideMap->containsOverride(attributeName))
```

Comment:
Should we first check if there any override exists for op? Op should be the key and then value is all overrides which should be applied to that Op?